src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgeryTokenGenerator.cs

```diff
@@ -60,16 +61,17 @@ public AntiforgeryToken GenerateCookieToken()
             };
 
             var isIdentityAuthenticated = false;
-            var identity = httpContext.User?.Identity as ClaimsIdentity;
+            var authenticatedIdentity = httpContext.User != null ? GetAuthenticatedIdentity(httpContext.User) : null;
```

Comment:
Remove the ternary and have GetAuthenticatedIdentity() return `null` when passed `null`.